vendor/k8s.io/kubernetes/test/e2e/apimachinery/aggregator.go

```diff
@@ -115,6 +118,33 @@ func cleanTest(client clientset.Interface, aggrclient *aggregatorclient.Clientse
 	_ = client.RbacV1().ClusterRoleBindings().Delete(context.TODO(), "wardler:"+namespace+":sample-apiserver-reader", metav1.DeleteOptions{})
 }
 
+func ensureSampleAPIUnregisteredFromAllRootAPIs(client clientset.Interface, namespace string) {
+	apiHelper, err := NewAPIEndpointHelper(client, namespace)
+	if err != nil && err == UnsupportedPlatformErr {
+		framework.Logf("This test runs on a platform that we don't know how to detect all instances of Kube API server." + " " +
+			"That means it might be flaky as there is no deterministic way of ensuring the sample server was unregistered from all replicas."  + " " +
+			"You can ignore this message if you are running on a platform with single api server.")
```

Comment:
I would tweak this message slightly and explain *how* we detect it.

```
This test runs with an API server that does not report endpoints to the kubernetes service in the default namespace. We cannot determine how many control plane API servers are running, which means we can't verify that all instances have stopped serving.  You can ignore this warning if you are running only a single API server instance.
```

I suspect we'd want to simply print this message in lots of places, so you can probably move it to NewAPIEndpointHelper and always print it.